core/src/main/scala/kafka/consumer/TopicCount.scala

```diff
@@ -104,6 +104,9 @@ private[kafka] object TopicCount extends Logging {
   def constructTopicCount(consumerIdString: String, filter: TopicFilter, numStreams: Int, zkUtils: ZkUtils, excludeInternalTopics: Boolean) =
     new WildcardTopicCount(zkUtils, consumerIdString, filter, numStreams, excludeInternalTopics)
 
+  override def equals(o: Any): Boolean = {
+    throw new KafkaException("can't use equals here")
+  }
```

Comment:
We are trying to ascertain whether we can get rid of `WildcardTopicCount#equals`, by adding a an equals method in the superclass that always throws an exception.  If the unit tests pass with this configuration, we should be able to get rid of the other one.